app/Http/Controllers/Playlist/PlaylistsController.php

```diff
@@ -834,33 +836,42 @@ public function completeItem(Request $request, $item_id)
             $playlist_item = PlaylistItems::where('id', $item_id)->first();
 
             if (!$playlist_item) {
-                return $this->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)->replyWithError('Playlist Item Not Found');
+                return $this
+                    ->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)
+                    ->replyWithError('Playlist Item Not Found');
             }
 
-            $user_plan = UserPlan::join('plans', function ($join) use ($user) {
-                $join->on('user_plans.plan_id', '=', 'plans.id')->where('user_plans.user_id', $user->id);
-            })
-                ->join('plan_days', function ($join) use ($playlist_item) {
-                    $join
-                        ->on('plan_days.plan_id', '=', 'plans.id')
-                        ->where('plan_days.playlist_id', $playlist_item->playlist_id);
-                })
-                ->select('user_plans.*')
-                ->first();
+            $user_plan = UserPlan::getByPlaylistIdAndUserId($playlist_item->playlist_id, $user->id);
 
             if (!$user_plan) {
                 return $this->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)->replyWithError('User Plan Not Found');
             }
 
-            $complete = $complete && $complete !== 'false';
+            $result = null;
 
-            if ($complete) {
-                $playlist_item->complete();
-            } else {
-                $playlist_item->unComplete();
+            try {
+                $complete = $complete && $complete !== 'false';
+
+                if ($complete) {
+                    $playlist_item->complete();
+                } else {
+                    $playlist_item->unComplete();
+                }
+
+                $result = $complete ? 'completed' : 'not completed';
+            } catch(QueryException $e) {
+                // Catch only the error code for a duplicate entry
+                if ($e->getCode() == MySQLErrorCode::DUPLICATE_ENTRY) {
```

Comment:
see note on processing above

Reply:
@bradflood , I have also been updated as you can see in the following line:

https://github.com/faithcomesbyhearing/dbp/pull/937/commits/296d9372d58d56d76d6f3881fdff2910acb12e39#diff-b2074cf830f20db0d05674376fd81b979af9a990752eb8e202431ca6729db2e4R863